packages/shared/src/components/sidebar/DiscoverSection.tsx

```diff
@@ -17,11 +29,34 @@ export function DiscoverSection({
   enableSearch,
   ...defaultRenderSectionProps
 }: DiscoverSectionProps): ReactElement {
+  const seoSidebar = useFeature(feature.seoSidebar);
+  const isV1Sidebar = seoSidebar === SeoSidebarExperiment.V1;
   const { checkHasCompleted, completeAction, isActionsFetched } = useActions();
   const hasCompletedCommentFeed =
     !isActionsFetched || checkHasCompleted(ActionType.CommentFeed);
-  const discoverMenuItems: SidebarMenuItem[] = [
-    {
+  const locationPush = useCallback((path: string) => {
+    const isExtension = checkIsExtension();
+
+    if (!isExtension) {
+      return null;
+    }
+
+    return window.location.assign(`${webappUrl}${path}`);
+    // router is unstable
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const discoverMenuItems: SidebarMenuItem[] = useMemo(() => {
+    const feeds = {
+      icon: (active: boolean) => (
+        <ListIcon Icon={() => <HotIcon secondary={active} />} />
+      ),
+      title: isV1Sidebar ? 'Explore' : 'Popular',
+      path: isV1Sidebar ? '/explore' : '/popular',
+      action: () => onNavTabClick?.(isV1Sidebar ? 'explore' : 'popular'),
+    };
```

Comment:
why have this separated here and not on the return below?

Reply:
Because of the change with the order.